src/Features/LanguageServer/Protocol/Handler/Diagnostics/Experimental/ExperimentalDocumentPullDiagnosticHandlerProvider.cs

```diff
@@ -15,19 +16,22 @@ internal class ExperimentalDocumentPullDiagnosticHandlerProvider : AbstractReque
 {
     private readonly IDiagnosticService _diagnosticService;
     private readonly IDiagnosticAnalyzerService _analyzerService;
+    private readonly EditAndContinueDiagnosticUpdateSource _editAndContinueDiagnosticUpdateSource;
 
     [ImportingConstructor]
     [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
     public ExperimentalDocumentPullDiagnosticHandlerProvider(
         IDiagnosticService diagnosticService,
-        IDiagnosticAnalyzerService analyzerService)
+        IDiagnosticAnalyzerService analyzerService,
+        EditAndContinueDiagnosticUpdateSource editAndContinueDiagnosticUpdateSource)
```

Comment:
this may need to be optional - I presume it doesn't exist in vscode?

Reply:
Is there a way to test?

I can't see how it wouldn't exist.. It's importing a concrete, not an interface, which is in Features and only imports one interface, for which the concrete is in LanguageServer.Protocol... so at the very least all of the DLLs would exist

Reply:
Ah - nevermind then - I didn't notice it was in Features.  So yeah it must be part of the composition.